src/quo2/components/wallet/account_card/view.cljs

```diff
@@ -45,72 +47,90 @@
-       (when metrics?
-         [rn/view {:style style/metrics-container}
-          [text/text
-           {:weight              :semi-bold
-            :size                :paragraph-2
-            :accessibility-label :metrics
-            :style               (style/metrics watch-only? theme)}
-           account-percentage]
-          (when (not empty-type?)
-            [:<>
-             [rn/view (style/separator watch-only? theme)]
+(def user-account
+  (let [pressed? (reagent/atom false)]
+    (fn [{:keys [state name balance percentage-value loading? amount customization-color type emoji
+                 metrics?
+                 theme on-press]}]
+      (let [watch-only?        (= :watch-only type)
+            empty-type?        (= :empty type)
+            account-amount     (if (= :empty state) "€0.00" amount)
+            account-name       (if (= :empty state) (i18n/label :t/Account 1) name)
+            account-percentage (if (= :empty state) "€0.00" percentage-value)]
+        (if loading?
+          [loading-view
+           {:customization-color customization-color
+            :type                type
+            :theme               theme
+            :metrics?            metrics?}]
+          [rn/pressable
+           {:on-press-in  #(reset! pressed? true)
+            :on-press-out #(reset! pressed? false)
+            :style        (style/card customization-color watch-only? metrics? theme @pressed?)
+            :on-press     on-press}
+           (when (and customization-color (not= :watch-only type))
+             [customization-colors/overlay
+              {:customization-color customization-color
+               :border-radius       16
+               :theme               theme
+               :pressed?            @pressed?}])
+           [rn/view {:style style/profile-container}
+            [rn/view {:style {:padding-bottom 2 :margin-right 2}}
+             [text/text {:style style/emoji} emoji]]
+            [rn/view {:style style/watch-only-container}
              [text/text
-              {:weight :semi-bold
-               :size   :paragraph-2
-               :style  (style/metrics watch-only? theme)} account-amount]
-             [rn/view {:style {:margin-left 4}}
-              [icon/icon :positive
-               {:color (colors/theme-colors (if watch-only? colors/neutral-50 colors/white-opa-70)
-                                            colors/white-opa-70
-                                            theme)
-                :size  16}]]])])])))
+              {:size   :paragraph-2
+               :weight :medium
+               :style  (style/account-name watch-only? theme)}
+              account-name]
+             (when watch-only? [icon/icon :reveal {:color colors/neutral-50 :size 12}])]]
+           [text/text
+            {:size   :heading-2
+             :weight :semi-bold
+             :style  (style/account-value watch-only? theme)}
+            balance]
+           (when metrics?
```

Comment:
Components should be smaller if we can do it, this greatly helps Reagent perform at its best.

For example, the whole hiccup inside `(when metrics? ...)` does not need to re-render when `name`, `balance`, `loading?`, `emoji`, `on-press` or `customization-color` change.

Reply:
Done